### ISSUE TITLE
Compile Sass into separate css file with AOT build

### ISSUE THREAD

**Issue by hariskabiljagic** [AngularClass/angular-starter#1797]
**Note: for support questions, please use one of these channels:** [Chat: AngularClass.slack](http://angularclass.com/member-join/) or [Twitter: @AngularClass](https://twitter.com/AngularClass)

* **I'm submitting a ...**
[ ] bug report
[ ] feature request
[ ] question about the decisions made in the repository

* **Do you want to request a *feature* or report a *bug*?**
I want to run a build:aot:prod and to compile sass files into a separate css file and link it in the head of the page.

* **What is the current behavior?**
Currently, build compiles sass into css and just insert separate (per component) css tags into the head of the page.
For build:dev and build:prod it works well if we remove the "include: [helpers.root('src', 'styles')]". 
But then build fails with build:aot:prod, because it complains about ExtractTextPlugin

Also, if we have "include: [helpers.root('src', 'styles')]" in webpack.prod.js config, then ExtractTextPlugin won't work.

* **If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem** via
https://plnkr.co or similar (you can use this template as a starting point: http://plnkr.co/edit/tpl:AvJOMERrnz94ekVua0u5).

* **What is the expected behavior?**
Expected behavior would be to compile all sass files from all component, merge them into one css file, minify it and have it as an asset which we can reference in the head of the page.

* **What is the motivation / use case for changing the behavior?**

* **Please tell us about your environment:**

- Angular version: 2.0.0-beta.X
- Browser: [all | Chrome XX | Firefox XX | IE XX | Safari XX | Mobile Chrome XX | Android X.X Web Browser | iOS XX Safari | iOS XX UIWebView | iOS XX WKWebView ]

* **Other information** (e.g. detailed explanation, stacktraces, related issues, suggestions how to fix, links for us to have context, eg. stackoverflow, gitter, etc)

**Comment by davidsekar** [comment]
Just a thought - While it should be possible to configure webpack to extract css from all component and bunde it to a CSS file, I wanted to highlight a feature you might miss - view encapsulation by Angular CLI. https://blog.thoughtram.io/angular/2015/06/29/shadow-dom-strategies-in-angular2.html

If you still like separate CSS file, then you can write separate SCSS for components in styles folder and import it to styles.scss(just a normal way).
A simpler alternative will

**Comment by hariskabiljagic** [comment]
I didn't actually missed that with ViewEncapsulation. Thing is, if we use ViewEncapsulation.Native for example to generate styles in the shadow root, imagine a repeated component like a button or some input...and if you have it on page repeated 10 times, it will generate shadow root 10 times. That html file might be a bit heavy in long term then. 
ViewEncapsulation.Emulated is really ugly...wouldn't like to use that in any case.
I use .None but it generate style tabs for each component in the head, and if we have 20 components on the page, we will have 20 style tabs in the head which is also really bad by my opinion.

Having separated styles for each component and not adding it as a styleUrls within component basically gets us back in past. And it violates the whole goal of a component :)

That is why I want to have the styles per component, and also some general styles for all site, and compile all that together in a separate css, minified and reference it in the head of the site.

I managed to do that for build:dev and build:prod but not for build:aot:prod

If anyone knows how to set it up, any help is well appreciated :)

**Comment by hariskabiljagic** [comment]
Solved this in unusual way, which I don't like 100% but it does what I want...
Basically when we create a component which has
component.ts
component.html
component.scss

we don't import scss in the component, we do an import of that file in the style.scss

That way it will work even with AOT, because it will generate one css  file containing all components styles and it won't add any style tabs in the head.

For this case I created a file
_components.scss where I imported all scss from all components in the project, and then imported that one in the style.scss.
So in style.scss we have
1. global styles
2. component specific styles.

Not perfect still I would say...although it works...

If anyone has better idea, let me know :)
